other/fun/bootstrap_node_info.py

```diff
@@ -42,57 +43,78 @@ def printHelp():
-MAX_MOTD_LENGTH = 256 # https://github.com/irungentoo/toxcore/blob/881b2d900d1998981fb6b9938ec66012d049635f/other/bootstrap_node_packets.c#L26
+# https://github.com/irungentoo/toxcore/blob/881b2d900d1998981fb6b9938ec66012d049635f/other/bootstrap_node_packets.c#L44
+VERSION_LENGTH = 4
+# https://github.com/irungentoo/toxcore/blob/881b2d900d1998981fb6b9938ec66012d049635f/other/bootstrap_node_packets.c#L26
+MAX_MOTD_LENGTH = 256
 
 MAX_INFO_RESPONSE_PACKET_LENGTH = PACKET_ID_LENGTH + VERSION_LENGTH + MAX_MOTD_LENGTH
 
 SOCK_TIMEOUT_SECONDS = 1.0
 
-sock = None
-
-if protocol == "ipv4":
-    sock = socket(AF_INET, SOCK_DGRAM)
-elif protocol == "ipv6":
-    sock = socket(AF_INET6, SOCK_DGRAM)
-else:
-    print("Invalid first argument")
-    printHelp()
-    sys.exit(1)
-
-sock.sendto(INFO_REQUEST_PACKET, (ip, port))
-
-sock.settimeout(SOCK_TIMEOUT_SECONDS)
-
-try:
-   data, addr = sock.recvfrom(MAX_INFO_RESPONSE_PACKET_LENGTH)
-except timeout:
-   print("The DHT bootstrap node didn't reply in " + str(SOCK_TIMEOUT_SECONDS) + " sec.")
-   print("The likely reason for that is that the DHT bootstrap node is either offline or has no info set.")
-   sys.exit(2)
-
-packetId = data[:PACKET_ID_LENGTH]
-if packetId != INFO_PACKET_ID:
-    print("Bad response, first byte should be", INFO_PACKET_ID, "but got", packetId, "(", data, ")")
-    print("Are you sure that you are pointing the script at a Tox DHT bootstrap node and that the script is up to date?")
-    sys.exit(3)
 
-version = int.from_bytes(data[PACKET_ID_LENGTH:PACKET_ID_LENGTH + VERSION_LENGTH], byteorder='big')
-motd = data[PACKET_ID_LENGTH + VERSION_LENGTH:].decode("utf-8")
-print("Version: " + str(version))
-print("MOTD:    " + motd)
-sys.exit(0)
+def main(protocol: str, host: str, port: int) -> None:
+    """Call the bootstrap node info RPC and output the response."""
+    if protocol == "ipv4":
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    elif protocol == "ipv6":
+        sock = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)
+    else:
+        print("Invalid first argument")
+        print_help()
+        sys.exit(1)
+
+    sock.sendto(INFO_REQUEST_PACKET, (host, port))
+
+    sock.settimeout(SOCK_TIMEOUT_SECONDS)
+
+    try:
+        data, _ = sock.recvfrom(MAX_INFO_RESPONSE_PACKET_LENGTH)
+    except socket.timeout:
+        print("The DHT bootstrap node didn't reply in " +
+              str(SOCK_TIMEOUT_SECONDS) + " sec.")
+        print("The likely reason for that is that the DHT bootstrap node "
+              "is either offline or has no info set.")
+        sys.exit(2)
+
+    packet_id = data[:PACKET_ID_LENGTH]
+    if packet_id != INFO_PACKET_ID:
+        print("Bad response, first byte should be " + str(INFO_PACKET_ID) +
+              " but got {packet_id}({data})".format({
+                  "packet_id": packet_id,
+                  "data": data,
+              }))
```

Comment:
`format()` [doesn't take a dict](https://docs.python.org/3/library/stdtypes.html#str.format).

```
$ yes | nc -lvu 127.0.0.1 33445 > /dev/null 2>&1 &
[1] 1644781
$ ./bootstrap_node_info.py
Traceback (most recent call last):
  File "bootstrap_node_info.py", line 116, in <module>
    main(
  File "bootstrap_node_info.py", line 94, in main
    " but got {packet_id}({data})".format({
KeyError: 'packet_id'
[1]+  Done                    yes | nc -lvu 127.0.0.1 33445 > /dev/null 2>&1
```

Either change it to `format_map()` or use named arguments.

Reply:
Done.